app/src/cose.h

```diff
@@ -131,25 +300,67 @@ namespace scitt::cose
       throw COSEDecodeError("Failed to decode protected header");
     }
 
-    if (header_items[ALG_INDEX].uDataType == QCBOR_TYPE_NONE)
+    if (header_items[PROFILE_INDEX].uDataType != QCBOR_TYPE_NONE)
     {
-      throw COSEDecodeError("Missing algorithm in protected header");
+      parsed.kid = cbor::as_string(header_items[PROFILE_INDEX].val.string);
     }
-    if (header_items[KID_INDEX].uDataType != QCBOR_TYPE_NONE)
+    if (header_items[ALG_INDEX].uDataType != QCBOR_TYPE_NONE)
     {
-      parsed.kid = cbor::as_string(header_items[KID_INDEX].val.string);
+      parsed.alg = header_items[ALG_INDEX].val.int64;
     }
     if (header_items[ISSUER_INDEX].uDataType != QCBOR_TYPE_NONE)
     {
       parsed.issuer = cbor::as_string(header_items[ISSUER_INDEX].val.string);
     }
+    if (header_items[CRIT_INDEX].uDataType != QCBOR_TYPE_NONE)
+    {
+      parsed.crit = std::vector<std::variant<int64_t, std::string>>();
+      QCBORItem critItem = header_items[CRIT_INDEX];
+      if (critItem.uDataType != QCBOR_TYPE_ARRAY)
+      {
+        throw COSEDecodeError(
+          "Value type of crit in protected header is not of type array");
+      }
```

Comment:
`t_cose` already checks this for us, since we specified `uDataType = QCBOR_TYPE_ARRAY`.